subs/pantry/src/Pantry/Types.hs

```diff
@@ -855,6 +879,18 @@ cabalSpecLatestVersion =
     CabalSpecV2_2 -> error "this cannot happen"
     CabalSpecV2_4 -> mkVersion [2, 4]
 
+data BuildFile = BFCabal !SafeFilePath !TreeEntry
+               | BFHpack !TreeEntry -- We don't need SafeFilePath for Hpack since it has to be package.yaml file
+  deriving (Show, Eq)
+
+isHPackBuildFile :: BuildFile -> Bool
```

Comment:
I would rather not have these two functions at all. Each one is only used once, and explicit pattern matching in those areas will likely have improvements on the code. For example, I don't like the idea of having a `ByteString` value present and then checking a `Bool` value to tell us whether to treat it as an hpack or cabal file.

Reply:
Remove both of them.